poseval/common/transformations.py

```diff
@@ -1,35 +1,60 @@
-class DataSpecs:
+import re
+
+import numpy as np
+
+
+class MalformedFenstringError(Exception):
+    pass
+
+
+class AlreadyWhiteError(Exception):
+    pass
 
+
+class DataSpecs:
     tensor6x8x8_sparse = 'tensor6x8x8'
     vector6x8x8_flat = 'vector6x8x8'
     tensor12x8x8_sparse = 'tensor12x8x8'
     vector12x8x8_flat = 'vector12x8x8'
 
     piece2layer_for_6x8x8_tensor = {
-        'p': 0,'P': 0,'n': 1,'N': 1,'b': 2,'B': 2,
-        'r': 3,'R': 3,'q': 4,'Q': 4,'k': 5,'K': 5
+        'p': 0, 'P': 0, 'n': 1, 'N': 1, 'b': 2, 'B': 2,
+        'r': 3, 'R': 3, 'q': 4, 'Q': 4, 'k': 5, 'K': 5
     }
 
     piece2value_for_6x8x8_tensor = {
-        'p': 1,'P': -1,'n': 1,'N': -1,'b': 1,'B': -1,
-        'r': 1,'R': -1,'q': 1,'Q': -1,'k': 1,'K': -1
+        'p': 1, 'P': -1, 'n': 1, 'N': -1, 'b': 1, 'B': -1,
+        'r': 1, 'R': -1, 'q': 1, 'Q': -1, 'k': 1, 'K': -1
     }
 
     piece2layer_for_12x8x8_tensor = {
-        'p': 0,'P': 1,'n': 2,'N': 3,'b': 4,'B': 5,
-        'r': 6,'R': 7,'q': 8,'Q': 9,'k': 10,'K': 11
+        'p': 0, 'P': 1, 'n': 2, 'N': 3, 'b': 4, 'B': 5,
+        'r': 6, 'R': 7, 'q': 8, 'Q': 9, 'k': 10, 'K': 11
     }
 
     piece2value_for_12x8x8_tensor = {
-        'p': 1,'P': 1,'n': 1,'N': 1,'b': 1,'B': 1,
-        'r': 1,'R': 1,'q': 1,'Q': 1,'k': 1,'K': 1
+        'p': 1, 'P': 1, 'n': 1, 'N': 1, 'b': 1, 'B': 1,
+        'r': 1, 'R': 1, 'q': 1, 'Q': 1, 'k': 1, 'K': 1
     }
 
+    col_num_dict = {'a': 0, 'b': 1, 'c': 2, 'd': 3, 'e': 4, 'f': 5, 'g': 6, 'h': 7}
+    num_col_dict = {value: key for key, value in col_num_dict.items()}  # reversed line_num_dict
```

Comment:
I think you could use `num_col = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']` and then instead of `col_num_dict` access it as `num_col.index('a')` - what do you think?